### PR TITLE
UpdateHint: create_if_not_exists (Allows clients to hint own items in other worlds)

### PR DESCRIPTION
**Pull request by NewSoupVi**
Basically, this is supposed to "supercede" https://github.com/ArchipelagoMW/Archipelago/pull/3523, and the concept of LocationScouts with create_as_hint in general.

I'm not sure if this should be its own new command called `CreateHint` instead.
If it were its own command, we could let the client provide a *list* of locations like with LocationScouts.
However, we would also have to maintain some stuff in two places. In my estimation, anything that you specify when creating a hint is also something you'd want to be able to update.

Lmk what you think.

### CODE REVIEW

**Comment by NewSoupVi** [comment]
On https://github.com/ArchipelagoMW/Archipelago/pull/3523, and also just from many interactions with users and world devs who have in-game hints, I can confidently say that this is a highly requested feature.

**Review by Jarno458 — COMMENTED** [review]
I would really like the ability of creating hints in bulk (like an array of locations to create hints for)
this is often useful for opening shops and sending hints for everything the player knows at that point

**Comment by NewSoupVi** [comment]
> does not allow to silently create hint
LocationScouts does not allow this either.

> Not broadcast hints a 2nd time
That is how this PR works, though.
The functionality that **is** removed in this PR is to rebroadcast hints over and over.

File: MultiServer.py [hunk 1]
```diff
@@ -1903,27 +1903,29 @@ async def process_client_cmd(ctx: Context, client: Client, args: dict):
...
+                if hint is None:
+                    # New hints are created with unspecified by default
+                    status_int = HintStatus.HINT_UNSPECIFIED
+                else:
+                    # If the hint already exists and no status was provided, there is no point to the packet
+                    return
+
             try:
-                status = HintStatus(status)
+                status = HintStatus(status_int)
```
Comment (Jarno458) [inline MultiServer.py:1928]: I would move the  try/catch for `status = HintStatus(status_int)` above the `hint = ctx.get_hint(client.team, location_player, location)` as it breaks the flow that there is a check for if hint is None right below it

**Review by Jarno458 — CHANGES_REQUESTED** [review]

**Comment by NewSoupVi** [comment]
Decided to go with a new command instead.

https://github.com/ArchipelagoMW/Archipelago/pull/4317

**Pull request closed without merging**